### PR TITLE
Update rocket example

### PR DESCRIPTION
**Pull request by Josef37**
- use values for moon ascent (was earth ascent)
- use values for real ascent module
- refactor to use easier syntax
- extract functions for readability
- add comments explaining equations
- label chart

### CODE REVIEW

**Comment by josdejong** [comment]
Looks like some neat improvements @Josef37, thanks!

I see you introduced a generic `${args}` in the expressions. I do understand that this gives less verbose code, but I prefer having the arguments explicitly, making it easier to read the code. I do like the function like `dgammadt` more clearly showing what parts it is built up from, but it is probably more straight forward to change functions like `gravity` and `angVel` to receive only the arguments that they actually use (instead of passing "everything")?

@ericman314 does it make sense to let the rocket take off from the moon instead of earth? Can you have a short look to see whether the expressions are still correct?

**Comment by Josef37** [comment]
> I see you introduced a generic `${args}` in the expressions. I do understand that this gives less verbose code, but I prefer having the arguments explicitly, making it easier to read the code. 

You're right on the readability. My reasoning was coming from that comment:  
`// It is important to maintain the same argument order for each of these functions.`
I think it's cleaner to enforce comments in code, but there is no problem in changing it back.

> [...] it is probably more straight forward to change functions like `gravity` and `angVel` to receive only the arguments that they actually use (instead of passing "everything")?

I would totally do that, I we chose to write the arguments out.

**Comment by ericman314** [comment]
Thank you for your interest in the rocket example--I never thought anyone would look at it again!

I appreciate the syntax refactoring, it's MUCH cleaner now.

> it is probably more straight forward to change functions like gravity and angVel to receive only the arguments that they actually use (instead of passing "everything")?

That is fine for `gravity` and `angVel`, but the other five functions must receive all five arguments or they won't work with `ndsolve`. I do agree with Jos on liking to see the args explicitly, rather than `${args}`. I understood it right away because of my prior familiarity with the code, but, defining an argument list using a string is not a very common pattern in JS in general so I worry it could be confusing for some.

> use values for moon ascent (was earth ascent)

I realize the description on the page says "Apollo Lunar Module". That was a mistake. This flight profile is actually that of an orbital launch of a rocket such as a Falcon 9. In a recent paper, we included this example but changed it to the lunar ascent so that our neglect of air resistance would not raise any questions. I think there is great value in showing how to stitch multiple stages of flight together in a single simulation, and I fear we would lose that if we changed to the single-stage lunar ascent. Perhaps there is a way to extend this example so a user can select either flight profile from a dropdown?

**Comment by Josef37** [comment]
I love that example (as someone with lots of hours in KSP :smile:).

Now the example includes an atmosphere and lots of cool charts. I used the official Falcon 9 values (and it gets close to the real flight profile). The only thing that's missing is earth's rotation, but that is just too much effort with this model.

There is one position code-wise I'm not safisfied with, but I'll leave it for today.

File: examples/browser/rocket_trajectory_optimization.html [hunk 1]
```diff
@@ -11,142 +12,290 @@
...
+        const resultMatrix = sim.evaluate(resultName)
+        const resultLength = resultMatrix.size()[0]
+        const startTime = time
+        time += dt * resultLength
+        const timeRange = math.range(startTime, time, dt)
+        const resultWithTime = math.concat(resultMatrix, math.reshape(timeRange, [resultLength, 1]))
+        result = result ? math.concat(result, resultWithTime, 0) : resultWithTime
+      }
+      return result
+    }
```
Comment (Josef37) [inline examples/browser/rocket_trajectory_optimization.html:238]: Here time is added to the results matrix.
The results don't know about time and there are different `dt`s for different stages. 
My best idea is to make it cleaner is this:
* To record the time in the solution, add another equation to the solver: `dtdt(...) = 1`
* Rename `tfinal` to something more precise (like `tDuration`, `simulationTime`, `tStage`, ...)

And maybe `concat` should be called only once at the end.

**Comment by ericman314** [comment]
Impressive. Most impressive. I really like it.

**Comment by Josef37** [comment]
Thanks. I'm glad you like it. 😃

I made a last revision (updating comments and extracting chart.js options). 
It should be ready to merge now. 
Feel free to make any edits you like.

**Review by ericman314 — APPROVED** [review]

**Comment by josdejong** [comment]
Wow, impressive Josef! Thanks. Though the example is _very_ extensive now, the code doesn't look intimidating at all, its easy to see what's going on. Nice job 👍

**Comment by josdejong** [comment]
The updated example is now published in v9.0.0 😄 

https://mathjs.org/examples/browser/rocket_trajectory_optimization.html

**Pull request merged**